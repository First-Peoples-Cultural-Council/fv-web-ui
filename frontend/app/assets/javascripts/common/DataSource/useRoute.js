import { useSelector } from 'react-redux'
function useRoute() {
  return {
    routeParams: useSelector((state) => state.navigation.route.routeParams),
  }
}
export default useRoute
