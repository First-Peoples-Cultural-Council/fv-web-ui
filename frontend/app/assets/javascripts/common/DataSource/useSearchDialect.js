import { useDispatch, useSelector } from 'react-redux'
import { searchDialectReset as _searchDialectReset } from 'providers/redux/reducers/searchDialect'
function usePortal() {
  const dispatch = useDispatch()
  const searchDialectReset = () => {
    const dispatchObj = _searchDialectReset()
    dispatch(dispatchObj)
  }
  return {
    computeSearchDialect: useSelector((state) => state.searchDialect.computeSearchDialect),
    searchDialectReset,
  }
}
export default usePortal
