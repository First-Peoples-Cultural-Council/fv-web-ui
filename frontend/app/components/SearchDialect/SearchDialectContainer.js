import React from 'react'
// import PropTypes from 'prop-types'
import SearchDialectPresentation from 'components/SearchDialect/SearchDialectPresentation'
import SearchDialectData from 'components/SearchDialect/SearchDialectData'

/**
 * @summary SearchDialectContainer
 * @version 1.0.1
 * @component
 *
 * @param {object} props
 *
 * @returns {node} jsx markup
 */
function SearchDialectContainer({ childrenUiSecondary, incrementResetCount }) {
  return (
    <SearchDialectData incrementResetCount={incrementResetCount}>
      {({ dialectClassName, formRefSearch, intl, onPressEnter, onSearch, searchTerm, searchStyle, onReset }) => {
        return (
          <SearchDialectPresentation
            childrenUiSecondary={childrenUiSecondary}
            dialectClassName={dialectClassName}
            formRefSearch={formRefSearch}
            intl={intl}
            onPressEnter={onPressEnter}
            onReset={onReset}
            onSearch={onSearch}
            searchTerm={searchTerm}
            searchStyle={searchStyle}
          />
        )
      }}
    </SearchDialectData>
  )
}
// PROPTYPES
// const { string } = PropTypes
SearchDialectContainer.propTypes = {
  //   something: string,
}

export default SearchDialectContainer
