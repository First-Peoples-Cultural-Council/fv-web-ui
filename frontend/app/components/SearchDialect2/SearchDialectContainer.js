import React from 'react'
// import PropTypes from 'prop-types'
import SearchDialectPresentation from 'components/SearchDialect2/SearchDialectPresentation'
import SearchDialectData from 'components/SearchDialect2/SearchDialectData'

/**
 * @summary SearchDialectContainer
 * @version 1.0.1
 * @component
 *
 * @param {object} props
 *
 * @returns {node} jsx markup
 */
function SearchDialectContainer({
  handleSearch: externalHandleSearch,
  resetSearch: externalResetSearch,
  searchDialectDataType: externalSearchDialectDataType,
  searchUi: externalSearchUi,
}) {
  return (
    <SearchDialectData
      handleSearch={externalHandleSearch}
      resetSearch={externalResetSearch}
      searchDialectDataType={externalSearchDialectDataType}
      searchUi={externalSearchUi}
    >
      {({
        dialectClassName,
        handleChangeSearchBySettings,
        handleEnterSearch,
        handleSearch,
        intl,
        partsOfSpeechOptions,
        resetSearch,
        searchByMode,
        searchBySettings,
        searchDialectDataType,
        searchDialectUpdate,
        searchMessage,
        searchNxqlQuery,
        searchTerm,
        searchType,
        searchUi,
      }) => {
        return (
          <SearchDialectPresentation
            dialectClassName={dialectClassName}
            handleChangeSearchBySettings={handleChangeSearchBySettings}
            handleEnterSearch={handleEnterSearch}
            handleSearch={handleSearch}
            intl={intl}
            partsOfSpeechOptions={partsOfSpeechOptions}
            resetSearch={resetSearch}
            searchByMode={searchByMode}
            searchBySettings={searchBySettings}
            searchDialectDataType={searchDialectDataType}
            searchDialectUpdate={searchDialectUpdate}
            searchMessage={searchMessage}
            searchNxqlQuery={searchNxqlQuery}
            searchTerm={searchTerm}
            searchType={searchType}
            searchUi={searchUi}
          />
        )
      }}
    </SearchDialectData>
  )
}
// PROPTYPES
// const { string } = PropTypes
SearchDialectContainer.propTypes = {
  //   something: string,
}

export default SearchDialectContainer
