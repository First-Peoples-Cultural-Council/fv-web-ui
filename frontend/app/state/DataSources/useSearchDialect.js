import { useDispatch, useSelector } from 'react-redux'
import {
  searchDialectReset as _searchDialectReset,
  searchDialectUpdate as _searchDialectUpdate,
} from 'reducers/searchDialect'

function useSearchDialect() {
  const dispatch = useDispatch()

  const searchDialectReset = () => {
    const dispatchObj = _searchDialectReset()
    dispatch(dispatchObj)
  }
  const searchDialectUpdate = (searchObj) => {
    const dispatchObj = _searchDialectUpdate(searchObj)
    dispatch(dispatchObj)
  }
  return {
    computeSearchDialect: useSelector((state) => state.searchDialect.computeSearchDialect),
    searchDialectReset,
    searchDialectUpdate,
  }
}

export default useSearchDialect
