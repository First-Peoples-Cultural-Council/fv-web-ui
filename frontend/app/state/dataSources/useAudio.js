import { useDispatch, useSelector } from 'react-redux'
import { setPlaying as _setPlaying, stopPlaying as _stopPlaying } from 'reducers/audio'
function useAudio() {
  const dispatch = useDispatch()
  return {
    lastPlayed: useSelector((state) => state.audio.lastPlayed),
    setLastPlayed: (src) => {
      dispatch(_setPlaying(src))
    },
    clearLastPlayed: () => {
      dispatch(_stopPlaying())
    },
  }
}
export default useAudio
