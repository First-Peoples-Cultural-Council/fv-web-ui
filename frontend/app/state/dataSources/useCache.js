import { useDispatch } from 'react-redux'
import { clearCacheItem as _clearCacheItem } from 'reducers/cache'

function usePortal() {
  const dispatch = useDispatch()

  const clearCacheItem = ({ id, reducer }) => {
    const dispatchObj = _clearCacheItem({ id, reducer })
    dispatch(dispatchObj)
  }

  return {
    clearCacheItem,
  }
}

export default usePortal
