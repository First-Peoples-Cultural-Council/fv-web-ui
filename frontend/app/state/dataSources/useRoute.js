import { useDispatch, useSelector } from 'react-redux'
import { setRouteParams as _setRouteParams } from 'reducers/navigation'

function useRoute() {
  const dispatch = useDispatch()

  const setRouteParams = (data) => {
    const dispatchObj = _setRouteParams(data)
    dispatch(dispatchObj)
  }

  return {
    routeParams: useSelector((state) => state.navigation.route.routeParams),
    navigationRouteSearch: useSelector((state) => state.navigation.route.search),
    setRouteParams,
  }
}

export default useRoute
