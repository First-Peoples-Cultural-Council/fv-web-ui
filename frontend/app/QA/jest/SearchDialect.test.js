/*
url driven: separate tests?

given props...
    - text filled
    - search type set
    - checkboxes clicked
    - part of speech selected
    - reset button displayed
    - feedback message displayed: separate tests?

Reset:
    - if query in url, changes url
    - if no query in url, clears input, resets to initial
    - if no text in search, button is hidden

Search:
    - if no text in search, do?
    - updates url with appropriate url queries

Alphabet:
    - no search ui
    - message displayed

Category:
    - no search ui
    - message displayed
*/
// Link.react.test.js
import React from 'react'
import renderer from 'react-test-renderer'
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}))

// jest.mock('dataSources/useSearchDialect', () => ({
//   ...jest.requireActual('dataSources/useSearchDialect'),
//   computeSearchDialect: jest.fn(),
//   searchDialectUpdate: jest.fn(),
//   searchDialectReset: jest.fn(),
// }))

// import * as useCustomHook from 'dataSources/useSearchDialect'
import * as useSearchDialect from 'dataSources/useSearchDialect'
const spyUseSearchDialect = jest.spyOn(useSearchDialect, 'default')
spyUseSearchDialect.mockReturnValue({
  computeSearchDialect: {
    searchByMode: 1,
    searchBySettings: {},
    searchMessage: '',
    searchNxqlSort: '',
    searchNxqlQuery: '',
    searchTerm: '',
    searchType: 1,
  },
  searchDialectUpdate: () => {},
  searchDialectReset: () => {},
})

import * as useRoute from 'dataSources/useRoute'
const spyUseRoute = jest.spyOn(useRoute, 'default')
spyUseRoute.mockReturnValue({
  routeParams: {
    letter: '',
    category: '',
    phraseBook: '',
  },
})
jest.mock('@material-ui/core/styles', () => ({
  /* eslint-disable-next-line */
  withStyles: (styles) => (component) => component,
}))

import SearchDialectContainer from 'components/SearchDialect2/SearchDialectContainer'

test('SearchDialect renders', () => {
  //   const fakeComputeSearchDialect = {
  //     searchByMode: 1,
  //     searchBySettings: {},
  //     searchMessage: '',
  //     searchNxqlSort: '',
  //     searchNxqlQuery: '',
  //     searchTerm: '',
  //     searchType: 1,
  //   }
  //   computeSearchDialect.mockResolvedValue(fakeComputeSearchDialect)
  const component = renderer.create(<SearchDialectContainer />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  /*
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onMouseEnter()
  // re-rendering
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onMouseLeave()
  // re-rendering
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  */
})
