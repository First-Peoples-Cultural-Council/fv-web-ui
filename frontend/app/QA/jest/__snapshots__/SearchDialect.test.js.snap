// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDialect renders 1`] = `
<div
  className="SearchDialect"
  data-testid="SearchDialect"
>
  
  <div
    className="SearchDialectForm SearchDialectForm--filtering"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Stop browsing Alphabetically
      </span>
    </button>
  </div>
</div>
`;
